src/main/resources/static/js/maps.js

```diff
@@ -0,0 +1,372 @@
+      console.log('cleared loader');
+    });
+  }
+  afterSearch = false;
+});
+
+let geolocation = new ol.Geolocation({
+  // enableHighAccuracy must be set to true to have the heading value.
+  trackingOptions: {
+    enableHighAccuracy: true,
+  },
+  projection: view.getProjection(),
+  tracking: true,
+});
+
+// handle maps error.
+geolocation.on('error', function (error) {
+  info.innerHTML = 'WARNING! ' + error.message;
+  info.style.display = '';
+  locationButton.disabled = true;
+});
+
+function showCurrentLocation() {
+  let coordinates = geolocation.getPosition();
+  map.getView().setCenter(coordinates);
+  map.getView().setZoom(12);
+}
+
+geolocation.once("change", function () {
+  showCurrentLocation();
+  newTennisLayer();
+});
+
+//adding button for centering on current location
+let locationButton = document.createElement('button');
+locationButton.innerHTML = "<i class='material-icons' data-toggle='tooltip' title='Current Location'" +
+    " style='color: white'>my_location</i>";
+
+locationButton.addEventListener('click', showCurrentLocation, false);
+
+let centerLocationElement = document.createElement('div');
+centerLocationElement.className = 'location ol-unselectable ol-control';
+
+centerLocationElement.appendChild(locationButton);
+
+let centerLocation = new ol.control.Control({
+  element: centerLocationElement,
+});
+map.addControl(centerLocation);
+//end of center on current location button
+
+//app starts searching courts after 2 seconds of inactivity on the map, to be changed if needed
+map.on("rendercomplete", debounced(2000, function () {
+  if(!afterSearch && !startedSearch && map.getView().getZoom() > 10) newTennisLayer();
+}))
+
+//temp functions for logging geolocation data in console
+function fillFormFieldsFromOsmFeature(feature){
+  let type = feature.id_.charAt(0).toUpperCase();
+  let id = feature.id_.split('/')[1];
+  let endpoint = 'https://nominatim.openstreetmap.org/lookup?osm_ids='+type+id+'&format=json&addressdetails=1';
+
+  fetch(endpoint)
+      .then(response => response.json())
+      .then(function (json) {
+        let object = JSON.parse(JSON.stringify(json));
+        const {house_number, country, postcode, state, road, city} = object[0].address;
+        courtHouseNumber.value = house_number !== undefined ? house_number : null;
+        courtAddress.value = road !== undefined ? road : null;
+        courtCity.value = city !== undefined ? city : null;
+        courtState.value = state !== undefined ? state : null;
+        courtCountry.value = country !== undefined ? country : null;
+        courtPostcode.value = postcode !== undefined ? postcode : null;
+        console.log(object);
+  });
+
+  let {email, name, phone, website} = feature.getProperties();
+  courtName.value = name !== undefined ? name : null;
+  courtEmail.value = email !== undefined ? email : null;
+  courtWebsite.value = website !== undefined ? website : null;
+  courtPhone.value = phone !== undefined ? phone : null;
```

Comment:
Są jakieś numery telefonu w OSM? Bo dla wszystkich dla których sprawdziłem to pole jest puste :(

Reply:
Są, ale bardzo rzadko. Jak już powiedziałem, dostajesz to co ktoś umieścił w OSM, więc raczej ciężko o dokładne informacje : / Nawet adres nie jest brany bezpośrednio z wyciągniętego feature'a tylko podpinam kolejne API, które wyciąga adres bazując na lokalizacji. Także, jak już nie raz wspomniałem, OSM raczej w celu nauki i zabawy niż faktycznej funkcjonalności.